pkg/db/src/implementations/version-1.2/books.ts

```diff
@@ -48,8 +50,12 @@ class Books implements BooksInterface {
 			.filter(([, book]) => book === undefined)
 			.map(([isbn]) => isbn);
 
-		Promise.all([...isbnsToFetch].map((isbn) => pluginManager.getBook(isbn))).then((res) => this.upsert(res));
-
+		const results = await Promise.all([...isbnsToFetch].map((isbn) => this.#pluginManagerInstance.getBook(isbn)));
+		// in case of disab;ed plugin, the results would be null
+		const filteredResults = results.filter((result): result is BookEntry => Boolean(result));
+		if (filteredResults.length) {
+			this.upsert(filteredResults);
+		}
```

Comment:
This can all be done in a single chain (just a style preference). It **shouldn't** be `await`ed (as we've discussed). Might as well feed the filtered array into `this.upsert` (It's a noop if array is empty).

Reply:
Additionally, upsert still uses `Promise.all`, I thought you've changed that to `bulkDocs` as agreed.